### ISSUE TITLE
Delay?

### ISSUE THREAD

**Issue by LefsFlare** [eccentricdevotion/GameModeInventories-2#1]
Sorry, I know that you will probably deny this issue but if possible I want to suggest that maybe a sort of delay could be implemented into the block listener? I have plugins like LogBlock installed and it really doesn't make sense how that one uses less CPU time than yours even if it actually does more database work.

At the very least, could you explain to me why it takes up so much CPU with just a few players online?

**Comment by eccentricdevotion** [comment]
No, I don't deny the issue, I just don't care that much, as I _never use the logging feature on my server._

Are you using SQLite or MySQL? If you are using SQLite, then database interaction cannot be queued for async writing, where as MySQL can. If you have `track_creative_place` enabled and are using SQLite, you need to switch.

If you are using MySQL then you should update to v2.7.4 (from the Jenkins server), and then report back as to whether it makes any difference (as there are some recent changes that should help).

**Comment by LefsFlare** [comment]
Alright, I get that you don't care that much about the logging but as of 1.8, your plugin is pretty much the only plugin that tracks creative inventory and creative block placement as well.

Regardless, with the public version released on bukkit, I've noticed that the Spigot 1.8 I ran this plugin on keeps crashing periodically and that's after I linked that version with a remote DB server. Any ideas?

**Comment by LefsFlare** [comment]
Alright, I get that you don't care that much about the logging but as of 1.8, your plugin is pretty much the only plugin that tracks creative inventory and creative block placement as well.

Regardless, with the public version released on bukkit, I've noticed that the Spigot 1.8 I ran this plugin on keeps crashing periodically and that's after I linked that version with a remote DB server. Any ideas?

**Comment by LefsFlare** [comment]
Alright, I get that you don't care that much about the logging but as of 1.8, your plugin is pretty much the only plugin that tracks creative inventory and creative block placement as well.

Regardless, with the public version released on bukkit, I've noticed that the Spigot 1.8 I ran this plugin on keeps crashing periodically and that's after I linked that version with a remote DB server. Any ideas?

**Comment by LefsFlare** [comment]
Alright, I get that you don't care that much about the logging but as of 1.8, your plugin is pretty much the only plugin that tracks creative inventory and creative block placement as well.

Regardless, with the public version released on bukkit, I've noticed that the Spigot 1.8 I ran this plugin on keeps crashing periodically and that's after I linked that version with a remote DB server. Any ideas?

**Comment by eccentricdevotion** [comment]
http://tardisjenkins.duckdns.org:8080/job/GameModeInventories/lastSuccessfulBuild/me.eccentric_nz.gamemodeinventories$GameModeInventories/

**Comment by LefsFlare** [comment]
Does it crash with a remote SQL DB linked? I don't want to keep on crashing my own server if it does.

**Comment by eccentricdevotion** [comment]
You'll have to test it... it's a completed recoded setup from the version available on Bukkit.

**Comment by LefsFlare** [comment]
Alright, I just installed the new version and I immediately regret it. Not including track_creative_place, the TPS of my server suffered intensely.

![Image](https://content.uploady.com/download/Spigot-Timings-Viewer.png?f=Spigot+Timings+Viewer.png&fid=fKE5VJbuNHH&p=Bw4NKFt4LuO&m=image%2Fpng&s=1&u=https%3A%2F%2Fwww.uploady.com%2Fdownload%2FfKE5VJbuNHH%2F67oZYOMCQ16LaXtt&tp=remote&t=1425143756&ex=172800&ip=1728786787&h=c0a4c699a8386bf5d6133d78b8da2901668e8662&width=934&height=300&type=thumb)

**Comment by LefsFlare** [comment]
Is remote MySQL impossible to do? Caching maybe?

**Comment by eccentricdevotion** [comment]
SQLite:
![screen shot 2015-03-01 at 9 42 18 am](https://cloud.githubusercontent.com/assets/1841924/6428023/6591b218-bff7-11e4-8184-57e32975ee3c.png)

MySQL:
![screen shot 2015-03-01 at 9 42 05 am](https://cloud.githubusercontent.com/assets/1841924/6428024/698ccd9e-bff7-11e4-8fe3-fe0da2d658a2.png)

TPS unaffected in either case

**Comment by LefsFlare** [comment]
How about REMOTE MySQLs? I have my main database server running offshore to offload the processing power but other plugins I have do not have problems with remote databases since they cache most of the results and only write to the database periodically
